pkg/envoy/xds/watcher.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium/pkg/envoy/api"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/sirupsen/logrus"
+)
+
+// ResourceWatcher watches and retrieves new versions of resources from a
+// resource set.
+// ResourceWatcher implements ResourceVersionListener to get notified when new
```

Comment:
ResourceVersionListener is not defined.

Reply:
Done. s/ResourceVersionListener/ResourceVersionObserver/